src/estimagic/optimization/tranquilo/sample_points.py

```diff
@@ -110,47 +117,207 @@ def _naive_sampler(
+    if n_points == 0:
+        return np.array([])
 
-    if bounds is not None and (bounds.lower is not None or bounds.upper is not None):
-        bounds = _get_effective_bounds(trustregion, bounds)
-        points = np.clip(
-            points,
-            a_min=bounds.lower,
-            a_max=bounds.upper,
-        )
+    if algo_options is None:
+        algo_options = {"stopping_max_iterations": 3}
 
+    effective_bounds = _get_effective_bounds(trustregion, bounds=bounds)
+
+    if existing_xs is not None:
+        # map existing points into unit space for easier optimization
+        existing_xs = (existing_xs - trustregion.center) / trustregion.radius
+
+    # start params
+    x0 = rng.normal(size=(n_points, n_params))
+    x0 = _project_onto_unit_hull(x0, ord=ord)
+
+    res = em.maximize(
+        criterion=_pairwise_distance_on_hull,
+        params=x0,
+        algorithm=algorithm,
+        criterion_kwargs={"existing_xs": existing_xs, "ord": ord},
+        lower_bounds=-np.ones_like(x0),
+        upper_bounds=np.ones_like(x0),
+        algo_options=algo_options,
+    )
+
+    points = _project_onto_unit_hull(res.params, ord=ord)
+    points = _map_into_feasible_trustregion(points, bounds=effective_bounds)
     return points
 
 
+# ======================================================================================
+# Helper functions
+# ======================================================================================
+
+
+def _pairwise_distance_on_hull(x, existing_xs, ord):  # noqa: A002
+    """Pairwise distance of new and existing points.
+
+    Instead of optimizing the distance of points in the feasible trustregion, this
+    criterion function leads to the maximization of the minimum distance of the points
+    in the unit space. These can then be mapped into the feasible trustregion.
+
+    Args:
+        x (np.ndarray): 2d array of internal points. Each value is in [-1, 1].
+        existing_xs (np.ndarray or None): 2d numpy array in which each row is an
+            x vector at which the criterion function has already been evaluated, that
+            satisfies -1 <= existing_xs <= 1.
+        ord (int): Type of norm to use when scaling the sampled points. For 2 we project
+            onto the hull of a sphere, for np.inf onto the hull of a cube.
+
+    Returns:
+        float: The criterion value.
+
+    """
+    x = _project_onto_unit_hull(x, ord=ord)
+
+    if existing_xs is not None:
+        sample = np.row_stack([x, existing_xs])
+        n_existing_pairs = len(existing_xs) * (len(existing_xs) - 1) // 2
+        slc = slice(0, -n_existing_pairs) if n_existing_pairs else slice(None)
+    else:
+        sample = x
+        slc = slice(None)
+
+    dist = pdist(sample) ** 2  # pairwise squared distances
+    dist = dist[slc]  # drop distances between existing points as they should not
+    # influence the optimization
+
+    crit_value = -logsumexp(-dist)  # smooth minimum
```

Comment:
This should have a scaling factor we can set and test in benchmarks.

Reply:
I chose the name `lse_scale` for logsumexp scale. Any better ideas for the argument name?

Reply:
The name should indicate the direction. It is the opposite of a smoothing factor. I would probably call it `hardness` and describe in the docstring that this refers to how closely the soft minimum approximates a hard minimum. Nobody who uses this even knows about what logsumexp is and why we use it here. 

And as always when we use that trick, link to this blogpost for further explanation: https://www.johndcook.com/blog/2010/01/20/how-to-compute-the-soft-maximum/